content/guide/property-system.md

```diff
@@ -0,0 +1,315 @@
+The examples of inherited properties are the`FontSize`, `FontWeight`, `Color`, etc.
+
+```ts
+import { Color, InheritedCssProperty, Style } from "@nativescript-core"
+
+export const selectedBackgroundColorProperty = new InheritedCssProperty<Style, Color>({
+    name: "selectedBackgroundColor",
+    cssName: "selected-background-color",
+    equalityComparer: Color.equals,
+    valueConverter: (v) => new Color(v)
+});
+selectedBackgroundColorProperty.register(Style);
+```
+#### Shorthand property
+To create a CSS property that can be applied using the shorthand syntax rules, use the [ShorthandProperty](https://docs.nativescript.org/api-reference/classes/shorthandproperty) class, passing it the [ShorthandPropertyOptions](#shorthandpropertyoptions-interface) object. For example, instead of setting margin using seperate rules for each side,
+```css
+.title {
+    margin-top:  0;
+    margin-right: 10;
+    margin-bottom: 0;
+    margin-left: 10;
+}
+```
+you can use one rule for all the sides.
+```css
+.title { 
+    margin: 0 10 0 10
+}
+```
+To create a shorthand property, use the CssPropery class to define all the properties to be shortened. Then return the shorthand with the `getter()` method of the ShorthandProperty class.
+<!-- TODO: Add an example -->
+```ts
+import { ShorthandProperty, Style } from "@nativescript-core"
+```
+### Creating a coercible property
+<!-- TODO: Add a proper explanation about when to create this property -->
+To create a coercible property use the [CoercibleProperty](https://docs.nativescript.org/api-reference/classes/coercibleproperty) class passing it a [CoerciblePropertyOptions](#co) object.
+
+```ts
+export const selectedIndexProperty = new CoercibleProperty<SegmentedBar, number>({
+  name: "selectedIndex", defaultValue: -1,
+  valueChanged: (target, oldValue, newValue) => {
+      target.notify(<SelectedIndexChangedEventData>{ eventName: SegmentedBar.selectedIndexChangedEvent, object: target, oldIndex: oldValue, newIndex: newValue });
+  },
+
+  // in this case the coerce value will change depending on whether the actual number of items
+  // is more or less than the value we want to apply for selectedIndex
+  coerceValue: (target, value) => {
+      let items = target.items;
+      if (items) {
+          let max = items.length - 1;
+          if (value < 0) {
+              value = 0;
+          }
+          if (value > max) {
+              value = max;
+          }
+      } else {
+          value = -1;
+      }
+
+      return value;
+  },
+
+  valueConverter: (v) => parseInt(v)
+});
+selectedIndexProperty.register(SegmentedBar);
+```
+Then when setting the property value, you call the `coerce()` method:
```

Comment:
Subsequently, when assigning a value to the property, invoke the coerce() method.